scw/client_option.go

```diff
@@ -172,7 +172,16 @@ func (s *settings) apply(opts []ClientOption) {
 func (s *settings) validate() error {
 	var err error
 	if s.token == nil {
-		return errors.New("no credential option provided")
+		return &ClientCredentialError{Type: clientCredentialError_NoOption}
+	}
+
+	if token, isToken := s.token.(*auth.Token); isToken {
+		if token.AccessKey == "" {
+			return &ClientCredentialError{Type: clientCredentialError_EmptyAccessKey}
```

Comment:
The only issue I can see here is people using terraform will see this error because their config was migrated from the v1 (no access_key included).

Reply:
We can get an `access_key` from a `secret_key` with the account API.

Reply:
Yes, the next question is where should it be done?

IMO we could do that when creating the SDK Client, if no access Key is provided, let's pull it.

Reply:
If we do it, it will be outside the validation and in another MR.